src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -415,10 +475,22 @@ private void TrackNullableStateForAssignment(BoundNode node, Symbol assignmentTa
                             valueIsNotNull;
                     }
 
-                    if (slot > 0 && targetType.TypeSymbol.IsAnonymousType && targetType.TypeSymbol.IsClassType() &&
-                        (value == null || targetType.TypeSymbol == value.Type))
+                    if (slot > 0)
                     {
-                        InheritNullableStateOfAnonymousTypeInstance(targetType.TypeSymbol, slot, GetValueSlotForAssignment(value), isByRefTarget);
+                        InheritDefaultState(slot);
```

Comment:
>InheritDefaultState(slot); [](start = 24, length = 26)

Is it possible that this will "erase" state that we are supposed to inherit below?

Reply:
Perhaps. I've added a `// PROTOTYPE` comment to follow up.